docs/source/development/cdf_guide.rst

```diff
@@ -0,0 +1,686 @@
+   epoch_data = [1, 2, 3]
+   epoch = xr.Variable(epoch_dims, epoch_data)
+   ds = xr.Dataset(data_vars={'data': data, 'epoch': epoch})
+   xarray_to_cdf(ds, 'hello.cdf')
+
+The xarray_to_cdf function converts an xarray.Dataset object into a CDF file.
+
+For IMAP, we will be using the following flags on xarray_to_cdf
+
+* datetime64_to_cdftt2000
+   * Working with numpy datetime64 objects for time is recommended because of the precision afforded, as well as the speed of calculations
+* istp
+   * Enabled by default.  This flag will perform some checks to verify that the function is able to make ISTP compliant CDF files from the Dataset object.
+* terminate_on_warning
+   * This flag ensures that the function terminates if there are any issues found, so we can avoid creating incomplete or non-compliant files.
+
+xarray_to_cdf with the above flags will perform the following steps -
+
+Verification
+-------------
+
+#. Verifies that required global attributes are present to meet ISTP compliance
+#. Verifies that variables have a VAR_TYPE attribute of either "data", "support_data", or "metadata"
+#. Verifies that all variables have the appropriate number of DEPEND_{i} attributes, based on the number of dimensions of the variable
+#. Verfies that the DEPEND_{i} attributes point to variables that are the size and shape expected to act as coordinates
+#. Verifies that each variable has the expected variable attributes to be ISTP compliant
+
+Conversion
+-----------
+
+#. Converts all variables with the word "epoch" in their name from datetime64 objects into the CDF native time format of CDF_TT2000 (nanoseconds since the year 2000)
+#. Converts all other data into one of the following CDF native formats - CDF_INT8, CDF_DOUBLE, CDF_UINT4, or CDF_CHAR
+#. Converts the attributes VALIDMIN, VALIDMAX, and FILLVAL to have the same data type as the data they are attached to
+   * For example, if you specify the attribute VALIDMIN=5 for a variable named Epoch, it will be converted to a CDF_TT2000 data type when written to the file
+#. Writes the Dataset attributes to the CDF file as the CDF's Global Attributes
+#. Writes the Dataset's variables to the CDF file
+   * Any variable with DEPEND_0 = Epoch as an attribute will be set to vary across records in the CDF file.  Otherwise, the data will be contained within a single record.
+#. Writes the attributes attached to the Dataset Variables to the file
+
+
+Shortcomings
+-------------
+While the above steps get a CDF file a large way towards ISTP compliance, there are several important caveats to take note of -
+
+* This code does not check the values *within* the attributes (except VAR_TYPE and DEPEND_{i}), only that the attributes exist!
+* This function does not put the "Epoch" variable as the first thing in the file, which was recommended (but not required) by the SPDF
+* This code creates variables in a row-major format.  Column-major is recommended (but not required) by the SPDF.
```

Comment:
Since we are just getting started, could we change to what they recommend?

Reply:
I can if I have time or if someone else wants to open a pull request on cdflib, but:

1) cdflib doesn't currently write column-major  
2) xarray_to_cdf loops through the variables in the xarray and writes them in whatever order depending on what it reads out of the dictionary first

I was somewhat surprised that those were recommended, since the software libraries abstract that stuff away.